### PR TITLE
Add scenario for MemoryCache

### PR DESCRIPTION
**Pull request by mikeharder**
@JunTaoLuo, @BrennanConroy, @Tratcher

### CODE REVIEW

**Comment by BrennanConroy** [comment]
:shipit:
Although do we want to add a scenario with reads and writes? What is the default expiration of a memory cache entry?

File: src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        public Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Path.StartsWithSegments(_path, StringComparison.Ordinal))
+            {
+                byte[] payload;
+                if (!_memoryCache.TryGetValue(_key, out payload))
+                {
+                    payload = Encoding.UTF8.GetBytes("Hello, World!");
+
+                    var cacheEntry = _memoryCache.CreateEntry(_key);
```
Comment (Tratcher) [inline src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs]: using, this doesn't actually add the entry without it.

File: src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    cacheEntry.Value = payload;
```
Comment (Tratcher) [inline src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs]: Add an expiration

**Comment by Tratcher** [comment]
There is no default expiration time, only a memory preasure trigger

File: src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs (hunk 2 above)
Comment (mikeharder) [inline src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs]: Yes, i will add a 1-second expiration for parity with the ResponseCaching scenario.

File: src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs (hunk 1 above)
Comment (mikeharder) [inline src/Benchmarks/Middleware/MemoryCachePlaintextMiddleware.cs]: Got it.  It wasn't clear how this was supposed to work from the interfaces.  I assumed `CreateEntry()` added it to the cache as well.

File: src/Benchmarks/Startup.cs [hunk 3]
```diff
@@ -115,6 +115,11 @@ public void ConfigureServices(IServiceCollection services)
                 }
             }
 
+            if (Scenarios.Any("MemoryCache"))
```
Comment (Tratcher) [inline src/Benchmarks/Startup.cs]: Shouldn't this match MemoryCachePlaintext?

Comment (mikeharder) [inline src/Benchmarks/Startup.cs]: It's future-proofing in case we add more scenarios which start with "MemoryCache".

**Comment by JunTaoLuo** [comment]
:ship::it:

**Pull request merged**